doc/src/manual/methods.md

```diff
@@ -515,6 +515,225 @@ julia> wait(schedule(t, 1))
 "definition for Int"
 ```
 
+## Design Patterns with Parametric Methods
+
+
+While complex dispatch logic is not required for performance or usability,
+sometimes it can be the best way to express some algorithm.
+Here are a few common design patterns that come up sometimes when using dispatch in this way.
+
+### Extracting the type parameter from a super-type
+
+
+Here is the correct code template for returning the element-type `T`
+of any arbitrary subtype of `AbstractArray`:
+
+```julia
+abstract AbstractArray{T, N}
+eltype(::Type{A}) where A<:AbstractArray{T} where T = T
```

Comment:
I don't think there are many short function definitions with more than one `where` so not sure how it is usually done then. In this case the following works and is quite nice:
```julia
eltype(::Type{A}) where {T,A<:AbstractArray{T}} = T
```